qlib/utils/serial.py

```diff
@@ -85,7 +86,7 @@ def to_pickle(self, path: Union[Path, str], dump_all: bool = None, exclude: list
         """
         self.config(dump_all=dump_all, exclude=exclude)
         with Path(path).open("wb") as f:
-            self.get_backend().dump(self, f)
+            self.get_backend().dump(self, f, protocol=C.dump_protocol_version)
```

Comment:
The backend may not be a pickle.
Maybe you can add  `Serializable.dump_kwargs`  for dumping and encourage users to override it.